### PR TITLE
Adds ID painting to the PDA painter

### PR DESCRIPTION
**Pull request by AffectedArc07**
The PDA painter is now the `color manipulator`, as it can now edit IDs.
This and the PDA list both have "friendly" names, instead of state names.

Sprite changes:
https://imgur.com/a/NJrcSKI

:cl: AffectedArc07
add: You can now skin IDs in the PDA painter
imageadd: The PDA painter sprite now has a glass slot to insert IDs to be painted
tweak: The "PDA painter" is now called "color manipulator"
tweak: You can now emag the PDA painter to paint any style of PDA or ID (apart from admin ones)
tweak: The PDA painter will now show "friendly" names for PDAs and IDs instead of icon state names
fix: You can now paint a PDA in any station PDA scheme
imageadd: The augment manipulator has its own sprite (Done by Quantum_M)
/:cl:

Why: We have needed a way to paint IDs since #525 and I feel that I could combine it into this. The code for this is also *slightly* less shitcode.

Huge thanks to flatty and dave for making this actually work.

### CODE REVIEW

File: code/game/machinery/PDApainter.dm [hunk 1]
```diff
@@ -144,6 +207,20 @@
 	else
 		to_chat(usr, "<span class='notice'>The [src] is empty.</span>")
 
+/obj/machinery/pdapainter/verb/ejectid()
```
Comment (FlattestGuitar) [inline code/game/machinery/PDApainter.dm]: These verbs are a crime against UX. Remove them and add an Alt+Click eject. Also put it in the description of the thing, in bold.

`\n<b>Alt-Click</b> to eject its contents!`

Comment (FlattestGuitar) [reply code/game/machinery/PDApainter.dm]: Also make them one proc, not two.

File: code/game/machinery/PDApainter.dm [hunk 2]
```diff
@@ -1,66 +1,106 @@
...
-		D.name = D.icon_state //PDAs don't have unique names, but using the sprite names works.
-
-		src.colorlist += D
-
 /obj/machinery/pdapainter/Destroy()
 	QDEL_NULL(storedpda)
+	QDEL_NULL(storedid)
 	return ..()
 
 /obj/machinery/pdapainter/on_deconstruction()
```
Comment (FlattestGuitar) [inline code/game/machinery/PDApainter.dm]: Once you make that single eject proc, use it here so as not to write the same code multiple times.

File: code/game/machinery/PDApainter.dm [hunk 3]
```diff
@@ -1,66 +1,106 @@
...
+	desc = "A machine able to color PDAs and IDs with ease. Insert an ID card or PDA and pick a color scheme."
 	icon = 'icons/obj/pda.dmi'
-	icon_state = "pdapainter"
+	icon_state = "coloriser"
 	density = TRUE
 	anchored = TRUE
 	var/obj/item/device/pda/storedpda = null
-	var/list/colorlist = list()
+	var/obj/item/card/id/storedid = null
+	var/pda_icons = list(
```
Comment (FlattestGuitar) [inline code/game/machinery/PDApainter.dm]: While the ID list is fine, the PDA list can be automated fairly sensibly. 
Put this in the init proc.
```
for(var/path in subtypesof(obj/item/device/pda)
    var/obj/item/device/pda/P = new path
    if(!P.hidden)
        pda_icons[P.name] = P.icon_state
    qdel(P)
```

File: code/game/machinery/PDApainter.dm (hunk 1 above)
Comment (FlattestGuitar) [reply code/game/machinery/PDApainter.dm]:
```
/obj/machinery/pdapainter/AltClick(mob/user)
* range check *
* incapacitated check *
* eject PDA *
* eject ID *
* mucho bene *
```

File: code/game/machinery/PDApainter.dm (hunk 3 above)
Comment (AffectedArc07) [reply code/game/machinery/PDApainter.dm]: Currently it goes by jobs, and I would rather it be standard to job than have PDA tacked onto it, as that was the reason to change it in the first place

File: code/game/machinery/PDApainter.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/game/machinery/PDApainter.dm]: There already are
`ejectid()`
`ejectpda()`

**Pull request merged**